hammertime/rules/redirects.py

```diff
@@ -0,0 +1,66 @@
+# hammertime: A high-volume http fetch library
+# Copyright (C) 2016-  Delve Labs inc.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+
+import copy
+
+from hammertime.http import Entry
+from hammertime.ruleset import Heuristics
+
+
+valid_redirects = (301, 302, 303, 307, 308)
+
+
+class FollowRedirects:
+
+    def __init__(self, *, max_redirect=15, stats):
+        self.max_redirect = max_redirect
+        self.engine = None
+        self.child_heuristics = Heuristics()
+        self.stats = stats
```

Comment:
Does not seem right to force providing the stats object to the heuristic just for this.